Makefile

```
@@ -92,6 +92,28 @@ ifeq ($(PROFILE),sso)
 AUTH_MODE                     := sso
 endif
 
+# Makefile managed tools
+TOOL_MOCKERY                := $(GOPATH)/bin/mockery
+TOOL_CONTROLLER_GEN         := $(GOPATH)/bin/controller-gen
+TOOL_GO_TO_PROTOBUF         := $(GOPATH)/bin/go-to-protobuf
+TOOL_PROTOC_GEN_GOGO        := $(GOPATH)/bin/protoc-gen-gogo
+TOOL_PROTOC_GEN_GOGOFAST    := $(GOPATH)/bin/protoc-gen-gogofast
+TOOL_PROTOC_GEN_GRPC_GATEWAY:= $(GOPATH)/bin/protoc-gen-grpc-gateway
+TOOL_PROTOC_GEN_SWAGGER     := $(GOPATH)/bin/protoc-gen-swagger
+TOOL_OPENAPI_GEN            := $(GOPATH)/bin/openapi-gen
+TOOL_SWAGGER                := $(GOPATH)/bin/swagger
+TOOL_GOIMPORTS              := $(GOPATH)/bin/goimports
+TOOL_GOLANGCI_LINT          := $(GOPATH)/bin/golangci-lint
+
+# npm bin -g will do this on later npms than we have
+TOOL_NODE_BINDIR            := $(shell npm config get prefix)/bin
+TOOL_CLANG_FORMAT           := /usr/local/bin/clang-format
+TOOL_MDSPELL                := $(TOOL_NODE_BINDIR)/mdspell
+TOOL_MARKDOWN_LINK_CHECK    := $(TOOL_NODE_BINDIR)/markdown-link-check
+TOOL_MARKDOWNLINT           := $(TOOL_NODE_BINDIR)/markdownlint
```

Since this is a simply expanded variable, it's going to execute this every time you run `make`, even if it's not actually used. This takes ~100ms for me, which isn't huge, but that can add up:
```
$ time npm config get prefix
/usr/local/share/nvm/versions/node/v20.19.0

real    0m0.114s
user    0m0.104s
sys     0m0.024s
```

The devcontainer uses `nvm` for handling Node versions, which makes this available in the `NVM_BIN` variable: https://github.com/nvm-sh/nvm?tab=readme-ov-file#environment-variables

So you could use optional assignment to make use of that and eliminate the unnecessary overhead, while still preserving compatibility for people without `nvm`:
```suggestion
NVM_BIN                     ?= $(shell npm config get prefix)/bin
TOOL_CLANG_FORMAT           := /usr/local/bin/clang-format
TOOL_MDSPELL                := $(NVM_BIN)/mdspell
TOOL_MARKDOWN_LINK_CHECK    := $(NVM_BIN)/markdown-link-check
TOOL_MARKDOWNLINT           := $(NVM_BIN)/markdownlint
```